modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/EncryptedFileIO.java

```diff
@@ -258,47 +233,13 @@ private void decrypt(ByteBuffer encrypted, ByteBuffer destBuf) throws IOExceptio
         assert encrypted.remaining() >= pageSize;
         assert encrypted.limit() >= pageSize;
 
-        int crc = FastCrc.calcCrc(encrypted, encryptedDataSize());
-
-        int storedCrc = 0;
-
-        storedCrc |= (int)encrypted.get() << 24;
-        storedCrc |= ((int)encrypted.get() & 0xff) << 16;
-        storedCrc |= ((int)encrypted.get() & 0xff) << 8;
-        storedCrc |= encrypted.get() & 0xff;
-
-        if (crc != storedCrc) {
-            throw new IOException("Content of encrypted page is broken. [StoredCrc=" + storedCrc +
-                ", calculatedCrc=" + crc + "]");
-        }
-
-        int keyId = encrypted.get() & 0xff;
-
-        encrypted.position(encrypted.position() - (encryptedDataSize() + 4 /* CRC size. */ + 1 /* key identifier. */));
-
-        encrypted.limit(encryptedDataSize());
+        int keyId = encrypted.get(encryptedDataSize() + 4 /* CRC size. */) & 0xff;
```

Comment:
```
        assert encrypted.remaining() >= pageSize;
        assert encrypted.limit() >= pageSize;
```
These conditions are checked by `EncryptionUtil#decrypt()`, so they can be removed here.